spec/Format/L.TileLayer.WMS.Format.js

```diff
@@ -40,4 +73,172 @@ describe('L.TileLayer.WMS.Format', function () {
   });
-});
+
+  describe('#getAvailable', function () {
+    it('returns parsed capabilities through specified \'done\' callback', function () {
+      // Stub AJAX util.
+      var originaAJAX = L.TileLayer.WMS.Util.AJAX;
+      L.TileLayer.WMS.Util.AJAX = sinon.spy(function (options) {
+        var responseText = '' +
+          '<?xml version="1.0" encoding="UTF-8"?>' +
+          '<WMS_Capabilities>' +
+          '  <Capability>' +
+          '    <Request>' +
+          '      <GetFeatureInfo>' +
+          '        <Format>format1</Format>' +
+          '        <Format>format2</Format>' +
+          '        <Format>format3</Format>' +
+          '      </GetFeatureInfo>' +
+          '    </Request>' +
+          '  </Capability>' +
+          '</WMS_Capabilities>';
+
+        return options.done(responseText);
+      });
+
+      // Prepare 'done' callback.
+      var done = sinon.spy(function (availableFormats) {});
+
+      // Call to 'getAvailable'.
+      L.TileLayer.WMS.Format.getAvailable({
+        url: 'http://test.ru',
+        done: done
+      });
+
+      // Retrieve available formats from 'done' callback arguments.
+      var availableFormats = done.getCall(0).args[0];
+
+      // Check that 'done' & 'AJAX' were called only once.
+      expect(done.calledOnce).to.be.equal(true);
+      expect(L.TileLayer.WMS.Util.AJAX.calledOnce).to.be.equal(true);
+
+      // Check retrieved elements.
+      expect(availableFormats).to.be.an('array');
+      expect(availableFormats).to.have.lengthOf(3);
+      expect(availableFormats).to.have.deep.property('[0]', 'format1');
+      expect(availableFormats).to.have.deep.property('[1]', 'format2');
+      expect(availableFormats).to.have.deep.property('[2]', 'format3');
+
+      // Restore stubbed AJAX util.
+      L.TileLayer.WMS.Util.AJAX = originaAJAX;
+    });
+
+    it('returns exception through \'fail\' callback if AJAX request succeed with text containing \'ServiceExceptionReport\'', function () {
+      var originaAJAX = L.TileLayer.WMS.Util.AJAX;
+
+      L.TileLayer.WMS.Util.AJAX = sinon.spy(function (options) {
+        var responseText = '' +
+          '<?xmlversion="1.0"?>' +
+          '<ServiceExceptionReport version="1.2.0">' +
+          '  <ServiceException code="999" locator="INSERTSTMT01">' +
+          '    parse error: missing closing tag for element WKB_GEOM' +
+          '  </ServiceException>' +
+          '</ServiceExceptionReport>';
+        return options.done(responseText);
+      });
+
+      var fail = sinon.spy(function (xhr) {});
+
+      // Call to 'getAvailable'.
+      L.TileLayer.WMS.Format.getAvailable({
+        url: 'http://test.ru',
+        fail: fail
+      });
+
+      expect(fail.calledOnce).to.be.equal(true);
+      expect(L.TileLayer.WMS.Util.AJAX.calledOnce).to.be.equal(true);
+      var str = fail.getCall(0).args[0].toString();
+      expect(str.indexOf("Error: Unable to parse specified 'xmlString' it isn't valid")).to.be.deep.equal(0);
```

Comment:
Message "Error: 999 - parse error: missing closing tag for element WKB_GEOM. " from the specified ServiceExceptionReport must be expected here, "Error: Unable to parse specified 'xmlString' it isn't valid" indicates that test responseText isn't a valid here (DOMParser can't parse it because in <?xmlversion="1.0"?> space is missing, it must be <?xml version="1.0"?>).